src/tests/Interop/ObjectiveC/ObjectiveCMarshalAPI/Program.cs

```diff
@@ -282,6 +289,13 @@ static unsafe void Validate_ReferenceTracking_Scenario()
             _Validate_ExceptionPropagation();
         }
 
+        [UnmanagedCallersOnly]
+        private static void BeforeThrowNativeException()
+        {
+            // This function is called from the exception propagation callback.
+            // It ensures that the thread was transitioned to preemptive mode.
```

Comment:
```suggestion
            // It ensures that the thread was transitioned to preemptive mode.
            GC.Collect();
```
Triggering GC verifies that the runtime and thread stack are in a good state.